src/llm_utils.py

```diff
@@ -0,0 +1,176 @@
+
+import os
+import json
+from google.genai import types
+from google import genai
+from typing import List, Dict
+
+def analyze_code_changes(diff_content: str) -> List[Dict]:
+    """
+    Analyze code changes using OpenAI's GPT model
+    Returns a list of review comments
+    """
+    API_KEY = os.getenv('GEMINI_API_KEY')
+
+    if not API_KEY:
+        raise ValueError("GEMINI_API_KEY is not set. Please add it to your environment variables.")
+
+    # Prepare the prompt for the LLM
+    prompt = f"""
+    Analyze the following code changes and provide review comments using EXACTLY this format:
+
+    FILE: [file-path]
+    LINE: [line-number]
+    COMMENT: [your comment]
+    SUGGESTION: [optional suggested code]
+
+    ---
```

Comment:
The prompt is very long and contains redundant information. The example formatting is good but repeating "IT IS A MUST!!!" seems unnecessary.

```suggestion
Shorten and clarify the prompt, removing redundant or overly emphatic phrases.
```